lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -1019,6 +1058,20 @@ def build_top_level_options(params):
         spec['InstanceInitiatedShutdownBehavior'] = params.get('instance_initiated_shutdown_behavior')
     if params.get('termination_protection') is not None:
         spec['DisableApiTermination'] = params.get('termination_protection')
+    if params.get('cpu_options') is not None:
+        cpu_opts = params.get('cpu_options')
+        spec['CpuOptions'] = {}
+        if cpu_opts.get('threads_per_core') is not None and cpu_opts.get('core_count') is not None:
+            if isinstance(cpu_opts.get('threads_per_core'), int):
```

Comment:
You're doing a lot of work here that could be done through a sub-argumentspec

Reply:
Done!